### PR TITLE
Deprecate WriteBatch.remove() and use the new style delete()

### PR DESCRIPTION
**Pull request by javeme**

### CODE REVIEW

File: java/src/main/java/org/rocksdb/AbstractWriteBatch.java [hunk 1]
```diff
@@ -85,6 +99,19 @@ public void delete(ColumnFamilyHandle columnFamilyHandle, byte[] key)
     delete(nativeHandle_, key, key.length, columnFamilyHandle.nativeHandle_);
   }
 
+  @Override
+  public void delete(ByteBuffer key) throws RocksDBException {
```
Comment (adamretter) [inline java/src/main/java/org/rocksdb/AbstractWriteBatch.java:103]: Can you mark the parameters as `final` please

**Review by adamretter — CHANGES_REQUESTED** [review]

File: java/src/main/java/org/rocksdb/AbstractWriteBatch.java (hunk 1 above)
Comment (javeme) [reply java/src/main/java/org/rocksdb/AbstractWriteBatch.java]: Done

**Review by adamretter — APPROVED** [review]

**Comment by adamretter** [comment]
@javeme Can you run `make format` please and append a commit

**Comment by adamretter** [comment]
@jay-zhuang @mrambacher Another one to merge please

**Pull request closed without merging**